overlord/snapstate/snapstate.go

```diff
@@ -1363,6 +1363,26 @@ func KernelInfo(st *state.State) (*snap.Info, error) {
 	return infoForType(st, snap.TypeKernel)
 }
 
+// AutoRefreshAssertions allows to hook fetching of important assertions
+// into the Autorefresh function.
+var AutoRefreshAssertions func(st *state.State, userID int) error
+
+// AutoRefresh is the wrapper that will do a refresh of all the installed
+// snaps on the system. In addition to that it will also refresh important
+// assertions.
+func AutoRefresh(st *state.State) ([]string, []*state.TaskSet, error) {
+	// FIXME: obtain it somehow?
```

Comment:
As we discussed, the FIXME can be dropped here. This is doing the right thing. It's the internal implementation of userID handling in SnapSetup vs. SnapState that is broken.